$OpenBSD: patch-drivers_pci_pcie_brcmstb_c,v 1.1 2021/10/21 12:58:29 jsg Exp $

revert 'pci: brcmstb: Cleanup controller state before handover'
85f3fddbcc2e8282f1c26dcef4684f59490d47ba

resetting pcie breaks pcie xhci on rpi4

Index: drivers/pci/pcie_brcmstb.c
--- drivers/pci/pcie_brcmstb.c.orig
+++ drivers/pci/pcie_brcmstb.c
@@ -577,24 +577,6 @@ static int brcm_pcie_probe(struct udevice *dev)
 	return 0;
 }
 
-static int brcm_pcie_remove(struct udevice *dev)
-{
-	struct brcm_pcie *pcie = dev_get_priv(dev);
-	void __iomem *base = pcie->base;
-
-	/* Assert fundamental reset */
-	setbits_le32(base + PCIE_RGR1_SW_INIT_1, RGR1_SW_INIT_1_PERST_MASK);
-
-	/* Turn off SerDes */
-	setbits_le32(base + PCIE_MISC_HARD_PCIE_HARD_DEBUG,
-		     PCIE_HARD_DEBUG_SERDES_IDDQ_MASK);
-
-	/* Shutdown bridge */
-	setbits_le32(base + PCIE_RGR1_SW_INIT_1, RGR1_SW_INIT_1_INIT_MASK);
-
-	return 0;
-}
-
 static int brcm_pcie_of_to_plat(struct udevice *dev)
 {
 	struct brcm_pcie *pcie = dev_get_priv(dev);
@@ -634,8 +616,6 @@ U_BOOT_DRIVER(pcie_brcm_base) = {
 	.ops			= &brcm_pcie_ops,
 	.of_match		= brcm_pcie_ids,
 	.probe			= brcm_pcie_probe,
-	.remove			= brcm_pcie_remove,
 	.of_to_plat	= brcm_pcie_of_to_plat,
 	.priv_auto	= sizeof(struct brcm_pcie),
-	.flags		= DM_FLAG_OS_PREPARE,
 };
